staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -108,6 +109,42 @@ func (c *cache) put(s string, a *Authenticator) *Authenticator {
 	return a
 }
 
+// sometimes rate limits how often a function f() is called. Specifically, Do()
+// will run the provided function f() up to threshold times every interval
+// duration.
+//
+// Note that this type is not thread-safe, so ya gotta do your own locking.
+type sometimes struct {
+	threshold int
+	interval  time.Duration
+
+	count  int       // times we have called f() in this window
+	window time.Time // beginning of current window of length interval
+}
+
+func newSometimes(threshold int, interval time.Duration) *sometimes {
+	return &sometimes{
+		threshold: threshold,
+		interval:  interval,
+	}
+}
```

Comment:
This function is not useful. You aren't doing any initialization.

Reply:
bd700ab99610a.